### DISCUSSION TITLE
Is nuxt 3 production ready?

### DISCUSSION THREAD

**Discussion by pavloniym** [nuxt/nuxt#19865] [Questions]
Hello!
Should I use nginx in front of nuxt 3 or nitro server can be exposed to real world directly?
What about RPS and highload?
Is nitro caches requests somehow?

Thank you!

**Comment by thejspro** [comment]
Nuxt 3 is not production ready. Still many, many bugs, I'm afraid :(

**Reply by renardsas** [reply]
A bit harsh as a comment. Yes there are bugs but Daniel and the rest of the team are very reactive. I use it in production and it works very well. Much more flexible and performant than Nuxt 2. Also, Nuxt 3 modules have really grown well these past few months.

**Comment by Jannchie** [comment]
The Nuxt team claims that version 3.0 is production-ready (https://nuxt.com/blog/v3). Based on my experience, there are still some bugs that need to be worked around. But the community is very responsive.

> What about RPS and highload?

Regarding the performance of the nitro, I found a benchmark test:
https://github.com/unjs/nitro/discussions/326#discussioncomment-6106292

> Should I use nginx in front of nuxt 3 or nitro server can be exposed to real world directly?

I'm not sure, but it may depend on your needs. In general, I personally think nginx is probably not needed, it introduces more complexity.
